torch/utils/optimizer.py

```diff
@@ -0,0 +1,26 @@
+"""
+This module contains utility method for mobile model optimization and lint.
+"""
+
+import io
+import torch
+
+
+def optimize_for_mobile(scripted_model):
+    """
+    Args:
+        scripted_model: An instance of torch script module with type of ScriptModule
+
+    Returns:
+        scripted_model: And optimized torch script module.
+    """
+    if not isinstance(scripted_model, torch.jit.ScriptModule):
+        raise TypeError(
+            'Got {}, but ScriptModule is expected.'.format(type(scripted_model)))
+
+    torch._C._jit_pass_optimize_for_mobile(scripted_model._c)
+    buffer = io.BytesIO()
+    torch.jit.save(scripted_model, buffer)
+    buffer.seek(0)
+    optimized_script_model = torch.jit.load(buffer)
```

Comment:
Curious why we have to save and load the model?  Does torch._C._jit_pass_optimize_for_mobile "enable" the pass, and it's only executed on save?

Reply:
Yes. @xcheng16, are you doing this because `_jit_pass_optimize_for_mobile` does inplace transformation of the script module? I think it probably is best for us to change that. Then you can just do `return torch._C._jit_pass_optimize_for_mobile(scripted_model._c)`

Reply:
torch._C._jit_pass_optimize_for_mobile(scripted_model._c) does not provide the inplace transformation of the script module, that is why I save and load again. I can check on _jit_pass_optimize_for_mobile method for why it is the case.